src/v/cloud_topics/core/serializer.cc

```diff
@@ -45,19 +37,30 @@ struct serializing_consumer {
         // Propagate to the output
         _output.append(std::move(hdr_iobuf));
         _output.append(std::move(rec_iobuf));
-        auto batch_size = _output.size_bytes() - offset;
-        _batches.emplace_back(num_records, base, batch_size, offset);
+        vassert(
+          _output.size_bytes() >= offset,
+          "Output size {} is smaller than offset {}",
+          _output.size_bytes(),
+          offset);
+        auto extent_size = _output.size_bytes() - offset;
+        _extents.emplace_back(extent_meta{
+          .id = object_id{},
+          .first_byte_offset = first_byte_offset_t(offset),
+          .byte_range_size = byte_range_size_t(extent_size),
+          .base_offset = base,
+          .committed_offset = kafka::offset(base() + num_records - 1),
```

Comment:
Why isn't this `base() + header.last_offset_delta`?

Reply:
> Why isn't this base() + header.last_offset_delta?

are you asking because it is clearer what's going on (avoids recomputation) or because the value would be different?